### DISCUSSION TITLE
How to get valid mesh/renderable touch point on motion event ?

### DISCUSSION THREAD

**Discussion by udayk07** [sceneview/sceneform-android#239] [Q&A]
While tap/ touch event action on any model in sceneform, it gives hit point on bounding box. If model is hollow in that touch point, how to get exact surface touch point in hit result ?

**Comment by ThomasGorisse** [comment]
I'm not completely sure to understand your question but maybe `node.localToWorldPosition()` is your answer.

**Reply by udayk07** [reply]
Sorry for abstract question. Here is an example case for your reference.

There is an sample model like below. In that, while ontap action on a model node, added new node ( black color node ) to the touch point on the model ( received from HitTestResult.getpoint() option ).
![Screenshot_20211213-222543_ZC_AR](https://user-images.githubusercontent.com/78372917/145856081-9963e95d-0fa3-4059-ad44-8055ed9568c1.jpg)

 
But, it seems the touch point is on the model's bounding box instead of actual mesh/model surface as it gives the result like below.

![Screenshot_20211213-222918_ZC_AR](https://user-images.githubusercontent.com/78372917/145856733-51bfda91-51b1-474a-b7de-cd0b4b587a27.jpg)

Is it possible to get mesh/model surface's world point on tap action ? ( so that the black color node always stick on the surface of the model )

**Reply by ThomasGorisse** [reply]
Good question.

It will request going inside the Sceneform Collision System.
Not sure that you really want to do so.

It works great but I rarely put my eyes on it.

If you are interested, you could ask in the Filament repo to know if it's possible with the new Pick API.

If that's the case, you could help integrating it inside sceneview-android.

**Reply by udayk07** [reply]
Thanks for the info ( Pick API ). Will check in Filament repo.

**Reply by udayk07** [reply]
@ThomasGorisse Pick api solves this offset problem. Pick api provides exact coordinates of model surface. Thanks for the suggestion
